e2e/src/pages/app.po.ts

```diff
@@ -128,43 +130,101 @@ export class AppPage {
   async drawOnImagePage() {
 
     await browser.executeScript(() => {
-      const imageElement = document.getElementsByTagName('mv-annotation-set')[0].childNodes[0];
+      // const imageElement = document.getElementsByTagName('mv-annotation-set')[0].childNodes[0];
+
+      const imageElement = document.getElementsByClassName('box-highlight')[0];
 
       const mouseDown = this.createMouseEvent('mousedown', 500, 500, 500, 500);
       const mouseMove = this.createMouseEvent('mousemove', 750, 750, 900, 900);
-      const mouseUp = this.createMouseEvent('mouseup', 750, 800, 750, 800);
+      const mouseUp =   this.createMouseEvent('mouseup', 750, 800, 750, 800);
 
       imageElement.dispatchEvent(mouseDown);
       imageElement.dispatchEvent(mouseMove);
       imageElement.dispatchEvent(mouseUp);
     });
   }
 
+
+
   async highLightTextOnPdfPage() {
-    await browser.executeScript(() => {
+    // const self  = this ;
+    // console.log( '~~~~~~~~~~~~~~ Self ...', self  );
+
+    await browser.executeScript( () => {
+
       const range = document.createRange();
       const matchingElement = document.getElementsByClassName('textLayer')[0].children[4];
       range.selectNodeContents(matchingElement);
       const sel = window.getSelection();
       sel.removeAllRanges();
       sel.addRange(range);
+
+      const mouseEvent = document.createEvent('MouseEvents');
+      mouseEvent.initMouseEvent(
+        'mouseup',
+        true,
+        true,
+        window,
+        1,
+        844,
+        497,
+        937,
+        403,
+        false,
+        false,
+        false,
+        false,
+        0,
+        null
+      );
+
+      //const mouseUp = 	mouseEvent;
+
+      const pageHandle = document.getElementsByClassName('textLayer')[0].children[4];
+      pageHandle.dispatchEvent(mouseEvent) ; // ('mouseup', 844,497,937,403)); //mouseUp Event
+
+
     });
 
-    this.getHighlightPopUp();
   }
 
   async getHighlightPopUp() {
     await browser.executeScript(() => {
-      const mousedown = document.createEvent('Event');
-      mousedown.initEvent('mousedown', true, true);
-      const mouseup = document.createEvent('Event');
-      mouseup.initEvent('mouseup', true, true);
-      const pageHandle = document.getElementsByClassName('pdfViewer')[0];
-      pageHandle.dispatchEvent(mousedown);
-      pageHandle.dispatchEvent(mouseup);
+
+      // const mousedown = document.createEvent('Event');
+      // mousedown.initEvent('mousedown', true, true);
+      //
+      // const mousemove = document.createEvent('Event');
+      // mousemove.initEvent('mousemove', true, true);
+      //
+      // const mouseup = document.createEvent('Event');
+      // mouseup.initEvent('mouseup', true, true);
+      //
+
+      const mouseDown = this.createMouseEvent('mousedown', 675, 405, 750, 412);
+      const mouseMove = this.createMouseEvent('mousemove', 750, 450, 900, 405);
+      const mouseUp =  this.createMouseEvent('mouseup', 844, 497, 937, 403);
```

Comment:
does this function actually work here since it doesn't work in the code above

Reply:
function just creates a highlight of phrase that in turns brings up the popup on the UI